pkg/service/export.go

```diff
@@ -65,3 +66,46 @@ func Export(clientProducer client.Producer, serviceNamespace, svcName string, st
 
 	return nil
 }
+
+func Exports(clientProducer client.Producer, serviceNamespace string, status reporter.Interface) error {
+	svcs, err := clientProducer.ForKubernetes().CoreV1().Services(serviceNamespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return status.Error(err, "Unable to list the Services in namespace %q", serviceNamespace)
+	}
+
+	if len(svcs.Items) == 0 {
+		status.Warning("No Services exist in target namespace")
+		return nil
+	}
+
+	for _, svc := range svcs.Items {
+		mcsServiceExport := &mcsv1a1.ServiceExport{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      svc.Name,
+				Namespace: serviceNamespace,
+			},
+		}
+
+		resourceServiceExport, err := resource.ToUnstructured(mcsServiceExport)
+		if err != nil {
+			return status.Error(err, "Failed to convert to Unstructured")
+		}
+
+		serviceExportGVR := gvr.FromMetaGroupVersion(mcsv1a1.GroupVersion, "serviceexports")
+
+		_, err = clientProducer.ForDynamic().Resource(serviceExportGVR).Namespace(serviceNamespace).
+			Create(context.TODO(), resourceServiceExport, metav1.CreateOptions{})
+		if err != nil {
+			if k8serrors.IsAlreadyExists(err) {
+				status.Success(fmt.Sprintf("Service %s already exported", svc.Name))
+				continue
+			}
+
+			return status.Error(err, "Failed to export Service")
+		}
+
+		status.Success(fmt.Sprintf("Service %s exported successfully", svc.Name))
```

Comment:
This could re-use `Export()` instead of effectively duplicating it — please change `Export()` to include the name of the service in its status messages, and then change `Exports()` to call `Export()`.

Reply:
This is a bit different. `Exports()` needs to list all svcs under the namespace. Maybe I can reuse Export in `for-range`

Reply:
Yes, my comment applied to the highlighted code only, not the `for` loop itself — I am well aware that `Exports()` and `Export()` are different :wink:.